Python/cp/Knuth_Morris_Pratt.py

```diff
@@ -0,0 +1,48 @@
+# Implementation of Knuth Morris Pratt Algorithm in Python
+def KMP_Search(pattern, text):
+    a = len(text)
+    b = len(pattern)
+    prefix = prefix_arr(pattern, b)
+  
+    initial = []
+    m = 0
+    n = 0
+  
+    while m != a:
+       
+        if text[m] == pattern[n]:
+            m += 1
+            n += 1
+      
+        else:
+            n = prefix[n-1]
+       
+        if n == b:
+            initial.append(m-n)
+            n = prefiX[n-1]
+        elif n == 0:
+            m += 1
+   
+    return initial
+def prefix_arr(pattern, b):
```

Comment:
Include appropriate Documentation (docstrings) for functions.

Reply:
Its proper as per the algorithm of KMP